### ISSUE TITLE
Upgrading 4.6.0 hasTable undefined error

### ISSUE THREAD

**Issue by steetstyle** [strapi/strapi#15603]
### Required System information

- Node.js version: v16.17.0
- NPM version: 8.15.0
- Strapi version: 4.6.0
- Database: PostgreSQL 14.6
- Operating system: macOS Monterey v12.6

### Describe the bug

After upgrading strapi from 4.5.6 to 4.6.0
```
error: Cannot read properties of undefined (reading 'hasTable')
TypeError: Cannot read properties of undefined (reading 'hasTable')
    at isLinkTableEmpty (/Users/***/node_modules/@strapi/database/lib/validations/relations/bidirectional.js:35:49)
    at validateBidirectionalRelations (/Users/***/node_modules/@strapi/database/lib/validations/relations/bidirectional.js:63:34)
    at validateRelations (/Users/***/node_modules/@strapi/database/lib/validations/relations/index.js:10:46)
    at validateDatabase (/Users/***/node_modules/@strapi/database/lib/validations/index.js:11:32)
    at Function.Database.init (/Users/***/node_modules/@strapi/database/lib/index.js:117:9)
    at Strapi.bootstrap (/Users/***/node_modules/@strapi/strapi/lib/Strapi.js:417:30)
    at Strapi.load (/Users/***/node_modules/@strapi/strapi/lib/Strapi.js:488:16)
    at async Strapi.start (/Users/***/node_modules/@strapi/strapi/lib/Strapi.js:217:9)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
````

I have deleted node_modules and yarn.lock and run ```yarn build```  but no luck 

### Additional context

Project dependencies in package.json:

```
"devDependencies": {},
  "dependencies": {
    "@_sh/strapi-plugin-ckeditor": "1.1.3",
    "@retikolo/drag-drop-content-types": "^1.3.4",
    "@strapi/plugin-i18n": "^4.6.0",
    "@strapi/plugin-upload": "^4.6.0",
    "@strapi/plugin-users-permissions": "^4.6.0",
    "@strapi/provider-email-sendgrid": "^4.6.0",
    "@strapi/provider-upload-aws-s3": "^4.6.0",
    "@strapi/strapi": "4.6.0",
    "@strapi/utils": "^4.6.0",
    "aws-sdk": "^2.1303.0",
    "minio-for-strapi-v4": "^1.0.7",
    "pg": "^8.8.0",
    "slugify": "^1.6.5",
    "strapi-plugin-populate-deep": "^2.0.0",
    "strapi-plugin-preview-button": "^1.1.2",
    "strapi-provider-upload-minio-ce": "^1.1.0"
  },
```

**Comment by dohoangtung** [comment]
Same issue!

**Comment by sam899** [comment]
Same for me

**Comment by sam899** [comment]
It seems this is a result of specifying a schema when using postgres on the `app/config/database.js` file as per this documentation https://docs.strapi.io/developer-docs/latest/setup-deployment-guides/configurations/required/databases.html#configuration-structure

If I remove the schema field in my config file, it works for me.

**Comment by Marc-Roig** [comment]
Hello, we are looking into this issue :) We will provide a ƒix asap.

**Comment by petersg83** [comment]
Thank you for the report! A fix has been created (#15702), it should be reviewed and merged for the next patch release (February 8th) :)
